Exiled.API/Features/Roles/Role.cs

```diff
@@ -202,6 +202,33 @@ protected Role(PlayerRoleBase baseRole)
         public virtual void Set(RoleTypeId newRole, SpawnReason reason, RoleSpawnFlags spawnFlags) =>
             Owner.RoleManager.ServerSetRole(newRole, (RoleChangeReason)reason, spawnFlags);
 
+        /// <summary>
+        /// Sets the player's <see cref="RoleTypeId"/>.
+        /// </summary>
+        /// <param name="newRole">The new <see cref="RoleTypeId"/> to be set.</param>
+        /// <param name="keepPosition">Indicates whether or not it should preserve the position after changing the role.</param>
+        /// <param name="keepInventory">Indicates whether or not it should preserve the inventory after changing the role.</param>
+        /// <param name="roleChangeReason">The <see cref="SpawnReason"/> defining why the player's role was changed.</param>
+        public void SetRole(RoleTypeId newRole, bool keepPosition = false, bool keepInventory = false, RoleChangeReason roleChangeReason = RoleChangeReason.RemoteAdmin)
+        {
+            RoleSpawnFlags roleSpawnFlags = RoleSpawnFlags.All;
+
+            if (keepPosition && keepInventory)
+            {
+                roleSpawnFlags = RoleSpawnFlags.None;
+            }
+            else
+            {
+                if (keepPosition)
+                    roleSpawnFlags = RoleSpawnFlags.AssignInventory;
+
+                if (keepInventory)
+                    roleSpawnFlags = RoleSpawnFlags.UseSpawnpoint;
+            }
```

Comment:
This thing will not work

Also use it like a flag not like an enum

Reply:
Unfortunately it can't be used like a flag. The value doesn't sum up. Regarding if it works or not, I would check it later.